applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_3D_law.cpp

```diff
@@ -0,0 +1,401 @@
+//************************************************************************************
+
+LinearIsotropicDamage3D::LinearIsotropicDamage3D(const LinearIsotropicDamage3D &rOther)
+    : ConstitutiveLaw(rOther)
+{
+}
+
+//********************************CLONE***********************************************
+//************************************************************************************
+
+ConstitutiveLaw::Pointer LinearIsotropicDamage3D::Clone() const
+{
+    LinearIsotropicDamage3D::Pointer p_clone(new LinearIsotropicDamage3D(*this));
+    return p_clone;
+}
+
+//********************************DESTRUCTOR******************************************
+//************************************************************************************
+
+LinearIsotropicDamage3D::~LinearIsotropicDamage3D()
+{
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool LinearIsotropicDamage3D::Has(const Variable<bool>& rThisVariable)
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        return true;
+    }
+    return false;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool LinearIsotropicDamage3D::Has(const Variable<double>& rThisVariable)
+{
+    if(rThisVariable == STRAIN_ENERGY){
+        return true;
+    }
+    return false;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool& LinearIsotropicDamage3D::GetValue(
+    const Variable<bool>& rThisVariable,
+    bool& rValue
+    )
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        rValue = mInelasticFlag;
+    }
+
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::InitializeMaterial(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues
+    )
+{
+    const double yield_stress = rMaterialProperties[YIELD_STRESS];
+    const double young_modulus = rMaterialProperties[YOUNG_MODULUS];
+    r_prev = yield_stress / std::sqrt(young_modulus);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::FinalizeSolutionStep(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // update of damage threshold
+    r_prev = r;
```

Comment:
Where `r` is defined?

Reply:
is a member defined in the .h file. I have renamed it to a more descriptive name though.